src/apple/system.rs

```diff
@@ -464,6 +464,48 @@ impl SystemExt for System {
         get_system_info(libc::KERN_OSTYPE, Some("Darwin"))
     }
 
+    fn get_long_os_version(&self) -> Option<String> {
```

Comment:
I'm a bit worried about this implementation: it means it'll require to be updated at every new macOS release, which is a bit suboptimal. Isn't there a way to get the version name by using a syscall instead?

Reply:
Unfortunately no, I could not find a `sysctl` that we can use. The best answer I found was: https://apple.stackexchange.com/questions/333452/how-can-i-find-the-friendly-name-of-the-operating-system-from-the-shell-term 

Happy to change if you can find one though!!

Reply:
Oh wow... It's terrifyingly bad. :rofl: